src/libs/actions/Session/index.js

```diff
@@ -72,24 +74,33 @@ function createAccount(login) {
  */
 function signOut() {
     Log.info('Flushing logs before signing out', true, {}, true);
+
     if (credentials && credentials.autoGeneratedLogin) {
-        // Clean up the login that we created
-        DeprecatedAPI.DeleteLogin({
+        const optimisticData = [
+            {
+                onyxMethod: 'set',
+                key: ONYXKEYS.SESSION,
+                value: null,
+            },
+            {
+                onyxMethod: 'set',
+                key: ONYXKEYS.CREDENTIALS,
+                value: null,
+            },
+        ];
+        API.write('LogOut', {
+            // Send current authToken because we will immediately clear it once triggering this command
+            authToken: NetworkStore.getAuthToken(),
```

Comment:
NAB, maybe kind of rare that this would happen - but feels like something that `API` should deal with by caching the authToken before setting `optimisticData`.